src/network/postgres/postgres_protocol_interpreter.cpp

```diff
@@ -83,7 +83,17 @@ Transition PostgresProtocolInterpreter::ProcessStartup(const common::ManagedPoin
       NETWORK_LOG_TRACE(db_name);
     }
   }
+
   auto oids = t_cop->CreateTempNamespace(context->GetConnectionID(), db_name);
+
+  uint32_t sleep_time = INITIAL_BACKOFF_TIME;
+  while ((oids.first == catalog::INVALID_DATABASE_OID || oids.second == catalog::INVALID_NAMESPACE_OID) &&
+         sleep_time <= MAX_BACKOFF_TIME) {
+    sleep(sleep_time);
+    oids = t_cop->CreateTempNamespace(context->GetConnectionID(), db_name);
+    sleep_time *= BACKOFF_FACTOR;
+  }
```

Comment:
Combine this together into a `do...while` loop instead of calling `CreateTempNamespace` in multiple locations.